### ISSUE TITLE
Failure to reproduce grounding-R1 7B results

### ISSUE THREAD

**Issue by shirley-wu** [Yan98/Grounding-R1#3]
Hi, thanks for the great work! I'm trying to reproduce the grounding-R1 7B results (acc = 50.1 on ScreenSpot-Pro), but haven't succeeded. Could you help me check what looks wrong? Thanks!

Below is my script and hyper-parameter setting. I run the experiments on one node with 8xH100
```bash
torchrun \
    --nproc_per_node 8 \
    src/grpo_grounding.py \
    --deepspeed local_scripts/zero3.json \
    --output_dir grounding/$RUN_NAME \
    --model_name_or_path "Qwen/Qwen2.5-VL-7B-Instruct"  \
    --dataset_name preprocessing/inp_full_no_wendy_lowe.json \
    --image_root "$HOME/data/grounding_dataset/image" \
    --max_prompt_length 1024 \
    --max_completion_length 128 \
    --num_generations 8 \
    --per_device_train_batch_size 2 \
    --freeze_vision_modules true \
    --reward_funcs accuracy \
    --beta 0 \
    --dataloader_num_workers 2 \
    --max_pixels $((4096 * 2160)) \
    --gradient_accumulation_steps 16 \
    --logging_steps 1 \
    --bf16 \
    --torch_dtype bfloat16 \
    --data_seed 42 \
    --report_to wandb \
    --gradient_checkpointing true \
    --attn_implementation flash_attention_2 \
    --num_train_epochs 2 \
    --run_name output/$RUN_NAME \
    --save_steps 10 \
    --save_total_limit 40 \
    --save_only_model false \
    --resume_from_checkpoint True
```
Based on my understanding, the global batch size should be 8 (# gpu) * 2 (bsz per device) * 16 (gradient accumulation) = 256, which is larger than 128

The accuracy of checkpoint-250/ is only 36.43; I also evaluated a few other checkpoints, and all of them are around 35. Below is my training curve, the variance seems larger than the figure you showed in your blog:

<img width="1192" alt="Image" src="https://github.com/user-attachments/assets/7bd0ea7e-20c1-41c1-b424-160ffe35d2b2" />

I would greatly appreciate it if you could help share any insights, thanks! Thank you again for the great work!

**Comment by Yan98** [comment]
Thank you for your interest in our work. We used the UI-TARS-1.5 7B model as the baseline for training our 7B model.

**Comment by shirley-wu** [comment]
Thank you so much for your reply!

**Comment by wwfnb** [comment]
> Thank you for your interest in our work. We used the UI-TARS-1.5 7B model as the baseline for training our 7B model. In your setting, the real batch size per gpu is 0.25, (per_device_train_batch_size/num_generations).

Hello, I found it on the uitars official website. They reported that the score of Uitars 1.5-7B on screenspot-pro was 49.6, but you measured it to be only 42. This seems to be a bit far off.

**Comment by shashacks** [comment]
> Thank you for your interest in our work. We used the UI-TARS-1.5 7B model as the baseline for training our 7B model. In your setting, the real batch size per gpu is 0.25, (per_device_train_batch_size/num_generations).

Could you explain more in detail? Why are you dividing with num_generations? Maybe for each GPU handles

**Comment by yunzhuzhang0918** [comment]
> Thank you for your interest in our work. We used the UI-TARS-1.5 7B model as the baseline for training our 7B model. In your setting, the real batch size per gpu is 0.25, (per_device_train_batch_size/num_generations).

I'm confused that why not multiply by the gradient accumulation? 
Could you please tell me what should the correct configuration be?
Look forward to your reply.
